data_model/src/events/data/events.rs

```diff
@@ -236,11 +320,31 @@ mod trigger {
     }
 }
 
+/// Has an origin whose change caused this event.
+pub trait Origin {
+    /// Type of the origin whose change caused this event.
+    type Origin: Identifiable;
```

Comment:
How about?

```suggestion
    type Origin: Identifiable::Id;
```

Reply:
Assuming your suggestion is literally

```suggestion
    type Origin = Identifiable::Id;
```
, _associated type defaults are unstable_

Reply:
I see, sorry